pkg/util/annotation.go

```diff
@@ -18,3 +25,54 @@ func MergeAnnotations(dst *unstructured.Unstructured, src *unstructured.Unstruct
 		MergeAnnotation(dst, key, value)
 	}
 }
+
+// GetWorkListByCRBName get works by ClusterResourceBinding name.
+func GetWorkListByCRBName(c client.Client, name string) ([]workv1alpha1.Work, error) {
+	workList := &workv1alpha1.WorkList{}
+	err := c.List(context.TODO(), workList)
```

Comment:
I'm afraid that it is potentially expensive that we list all RBs every time and range them to filter the matched ones when we have a lot of RBs. What about only moving names to annotations and leave namespace in labels? So we could list work by namespace label first. That will narrow work list.

Reply:
Sounds good.

Reply:
But separating namespaces and namespaces can be confusing. How do you think @RainbowMango

Reply:
Can we have both the `label` and `annotation`? E.g.
```diff
diff --git a/pkg/apis/work/v1alpha1/well_known_labels.go b/pkg/apis/work/v1alpha1/well_known_labels.go
index 9cc3d813..657ad8cc 100644
--- a/pkg/apis/work/v1alpha1/well_known_labels.go
+++ b/pkg/apis/work/v1alpha1/well_known_labels.go
@@ -1,6 +1,15 @@
 package v1alpha1

 const (
+       // ResourceBindingReferenceKey is the key of ResourceBinding object.
+       // It is usually a unique hash value of ResourceBinding object's name and intended to be added to the Work object.
+       // It will be used to retrieve all Works objects that derived from a specific ResourceBinding object.
+       ResourceBindingReferenceKey = "resourcebinding.karmada.io/key"
+       // ClusterResourceBindingReferenceKey is the key of ClusterResourceBinding object.
+       // It is usually a unique hash value of ClusterResourceBinding object's name and intended to be added to the Work object.
+       // It will be used to retrieve all Works objects that derived by a specific ClusterResourceBinding object.
+       ClusterResourceBindingReferenceKey = "clusterresourcebinding.karmada.io/key"
+
        // ResourceBindingNamespaceLabel is added to objects to specify associated ResourceBinding's namespace.
        ResourceBindingNamespaceLabel = "resourcebinding.karmada.io/namespace"
```

The label is used to retrieve `Work` by `RB/CRB` and the `annotation` is used to retrieve `RB/CRB` by `Work`.

Reply:
That's greate. How do you think @Garrybest ?

If that's ok, i am going to modify with this command.

Reply:
So we have both annotations and labels of resourcebinding in work? And the value is a unique hash value? I don't understand why annotations still need to be kept if we keep the labels. Could you give a example?

Reply:
Good question. 
https://github.com/karmada-io/karmada/blob/1947cf219979c728f1aa63fd185b33efd821b1de/pkg/controllers/binding/binding_controller.go#L120-L121

The binding controller wants to enqueue `Binding` once the relevant `Work` is changed, in order to such as, grab status.

Reply:
OK, let's do it.